### PR TITLE
[WIP] 1.3 new UI

### PR DESCRIPTION
**Pull request by lugi0**
Signed-off-by: Luca Giorgi <lgiorgi@redhat.com>

### CODE REVIEW

**Comment by lugi0** [comment]
Currently blocked by a bug, will test via jenkins job once it's updated

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (vasukulkarni) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:27]: need to handle the code for 1.2.0-2

File: tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot
Comment (vasukulkarni) [inline tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot:69]: does it need a space between `label_selector= and From=`

Comment (lugi0) [reply tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot:69]: It does not! :D

File: libs/Helpers.py
Comment (jgarciao) [inline libs/Helpers.py:15]: Does it work with versions like "1.2.0-2"? If the "-2" part is a problem maybe you could use the semver library to parse it and to make the comparisons
https://python-semver.readthedocs.io/en/latest/usage.html

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (jgarciao) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:23]: This two lines could be condensed in one like this:
```
Before:
 ${ver} =  Get RHODS version
 ${version-check} =  GTE  ${ver}  1.3.0
 IF  ${version-check}==True

After:
${version-check} = RHODS Version Is Greater Or Equal Than  1.3.0
IF  ${version-check}==True
```

File: libs/Helpers.py
Comment (lugi0) [reply libs/Helpers.py:15]: From my testing, it does. (1,2,0,2) evaluates as smaller than (1,3,0). To the best of my knowledge tuple comparisons are implemented positions by position

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (lugi0) [reply tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot]: Implemented in new commit

File: libs/Helpers.py
Comment (lugi0) [reply libs/Helpers.py:15]: Implemented semver in new commit

**Comment by lugi0** [comment]
I've had to move from openshiftcli to running the oc command directly to fix a test failure due to a timing issue

File: libs/Helpers.py
Comment (jgarciao) [inline libs/Helpers.py:14]: Not too important but how about creating a new file called version_comparer.py (or similar) an name this method `is_gte`?

File: tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot
Comment (jgarciao) [inline tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot:74]: This command returns an empty line in ods-qe-test2. Maybe you want to replace it for this one:
`oc get csv -n redhat-ods-operator | grep "rhods-operator" | awk '{print $1}' | sed 's/rhods-operator.//'`

Have you tried getting the version in PSI clusters? Does it work well there too?

File: libs/Helpers.py
Comment (lugi0) [reply libs/Helpers.py:14]: I'm not sure it would provide any benefit?
I think for now we can keep the small helper functions in one file, and eventually break them up later on

File: tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot
Comment (lugi0) [reply tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot:74]: The command works on PSI for sure, I'll have to investigate why it's not working in that osd cluster

Comment (lugi0) [reply tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot:74]: For some reason the formatting of the output of 
`oc get csv -n redhat-ods-operator | grep "rhods-operator"`
is formatted differently between PSI and OSD. 

Screenshot from 2021-11-26 11-41-49
Screenshot from 2021-11-26 11-41-23

your variation of the command works on both, so I'll update to that

Comment (lugi0) [reply tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot]: Updated in latest commit

Comment (jgarciao) [reply tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot]: Thanks!

File: libs/Helpers.py
Comment (jgarciao) [reply libs/Helpers.py:14]: ok!

**Review by jgarciao — APPROVED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Comment by vasukulkarni** [comment]
Please feel free to merge after testing with current build.

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (bdattoma) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:137]: I think you can move this line of code outside the IF statement because duplicate

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (bdattoma) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:27]: maybe during refactoring we should change these absolute xpaths

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (lugi0) [reply tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:116]: I don't exactly remember why it wasn't working with the `Page Should Contain`

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (lugi0) [reply tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:27]: We probably won't need the `ELSE` block anymore after 1.6, we need to support N and N-1 only

**Review by jgarciao — DISMISSED** [review]
LGTM

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (bdattoma) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:23]: since we are going to support the N-1 version, maybe it is worth to store the version in a common variable, so we don't need to change everywhere manually every release that requires a customization

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Comment (bdattoma) [reply tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:116]: okay, seems strange

**Review by bdattoma — DISMISSED** [review]

**Comment by bdattoma** [comment]
LGTM. could you run a jenkins job before merging?

**Comment by lugi0** [comment]
I think we will run smoke with this PR before merging

**Comment by jgarciao** [comment]
I believe the keyword `User Is Allowed` is failing in 1.5.0 because it's using `xpath:/html/body/div[1]/form/div/div/div[2]`, which I think doesn't exist any more.

Maybe you could add a new keyword "Wait Until Spawner Is Loaded" or similar and use it in `User Is Allowed` and `Select Notebook Image`

**Comment by lugi0** [comment]
@jgarciao quick and dirty fix for the keyword for now, we can think of a way to improve it later

**Review by bdattoma — APPROVED** [review]

**Review by tarukumar — APPROVED** [review]
LGTM

**Review by aloganat — APPROVED** [review]
LGTM

**Pull request merged**
